### PR TITLE
Sprint 24

### PR DESCRIPTION
**Pull request by ericsorenson**
# Sprint 24

Reminder: update policies before a deployment for ElasticSearch.

## [1282 Judge: Batch download cases/document](https://github.com/flexion/ef-cms/issues/1282)
image

## [2470 Form/Error Clean up](https://github.com/flexion/ef-cms/issues/2470)
image

## [2019 QC Box Cleanup](https://github.com/flexion/ef-cms/issues/2019)
image

## [2828 Court User: Search for Case by Docket Number or Keyword](https://github.com/flexion/ef-cms/issues/2828)
image

## [981 Petitions Clerk: Summary of Orders to be Generated During QC](https://github.com/flexion/ef-cms/issues/981)
image

## [996 Epic 3 Clean-up](https://github.com/flexion/ef-cms/issues/996)
image

## [2875 Court Users: Ability to sign a court-issued document](https://github.com/flexion/ef-cms/issues/2875)
image

### CODE REVIEW

**Comment by waldoj** [comment]
Here's the review checklist that I'll work my way through in reviewing this PR:

- [ ] begin by deleting any existing copy of the Git repository and cloning a fresh copy, running `docker system prune` to remove any cached artifacts from any prior code review, and running `./reset-dependencies.sh` to update Node packages
- [ ] review the pull request itself, to get oriented
	- [ ] read the description of the pull request, which should summarize the changes made
	- [ ] read through every task on the Scrum board that's encompassed by this pull request
	- [ ] read the description of the commits that comprise the pull request
- [ ] stand up the site locally
	- [ ] test all functionality in all major browsers, emphasizing the functionality that this pull request addresses
		- [ ] for internal Court functionality, perform the most thorough testing in Chrome, though also test in Edge and Firefox
		- [ ] for public-facing functionality, test in browsers consistent with [public browser use data](https://analytics.usa.gov/), including Internet Explorer 11
		- [ ] test in Mobile Safari and Mobile Chrome, with the caveat that not all internal Court functionality will be necessary on these platforms
	- [ ] use an automated audit tool for code quality and practices (recommended: [Chrome DevTools](https://developers.google.com/web/tools/chrome-devtools/), aka [Lighthouse](https://chrome.google.com/webstore/detail/lighthouse/blipmdconlkpinefehnmjammfjpmpbjk))
		- [ ] look at efficiency of page loads, asset sizes, HTTP connection management, etc.
	- [ ] review for accessibility
		- [ ] use an automated audit tool, such as Chrome Audit or aXe
		- [ ] navigate site only with the keyboard
		- [ ] use VoiceOver or Narrator to navigate the site with audio only, with the display turned off
		- [ ] manually test anything that pa11y cannot test automatically (e.g., contrast of text over images)
- [ ] review static code analysis results in SonarCloud
- [ ] run a security audit of dependencies (`npm audit`) to ensure that there are no vulnerabilities that will be deployed to production (as opposed to vulnerabilities that only have an impact on the development environment)
- [ ] examine OWASP ZAP output in `docs/`, to ensure that any errors are known to be false positives or have been previously declared to be acceptable
- [ ] skim all new code, in the context of existing code, [looking for problems](https://github.com/ustaxcourt/ef-cms/blob/staging/docs/CODE_REVIEW.md#what-we-look-for) (knowing that the vast majority of new code will be covered by tests)
- [ ] review all tests
	- [ ] methodically review all new tests for correctness, quality of naming
	- [ ] look at code coverage of tests in SonarCloud
	- [ ] determine what code isn’t tested, review that rigorously
- [ ] review documentation to ensure that it matches changes
- [ ] provide comments on the pull request on GitHub, as necessary
	- [ ] for comments that are specific to a particular line of code, comment on those specific lines
	- [ ] for comments that are more general, attach the comment to a random line in `README.md` (as opposed to commenting on the pull request itself), to be able to use GitHub's ability to thread discussions on those comments
- [ ] for each feature-level bug (i.e., it’s working as designed, but designed wrong), open a new issue and put it in the backlog

**Comment by waldoj** [comment]
Because GitHub lacks effective pull-request-level feedback mechanisms that must be addressed before a pull request can be merged, I'll be using `.dockerignore` as the place to attach PR-level comments.

File: .dockerignore [hunk 1]
```diff
@@ -11,10 +11,14 @@ web-api/.dynamodb
 web-api/.nyc_output
 web-api/.serverless
 web-api/coverage
-web-api/.cache
```
Comment (waldoj) [inline .dockerignore]: Because GitHub lacks effective pull-request-level feedback mechanisms that must be addressed before a pull request can be merged, I am using this file as the place to attach PR-level comments.

File: .dockerignore [hunk 2]
```diff
@@ -11,10 +11,14 @@ web-api/.dynamodb
...
 web-api/dist
+web-api/.cache
```
Comment (waldoj) [inline .dockerignore]: I have been unable to batch download cases, as covered by [story 1282](https://github.com/flexion/ef-cms/issues/1282). This is true across browsers, in any session that has cases. This is while running the site locally, using `docker-run.sh`. (@rachaelparris has been able to replicate this, luckily.) Please address this problem.

Comment (ericsorenson) [reply .dockerignore]: Updated.

File: policy.json.tpl [hunk 3]
```diff
@@ -118,8 +129,15 @@
...
         "dynamodb:UpdateTable",
+        "dynamodb:UpdateTimeToLive",
+        "dynamodb:CreateGlobalTable",
         "dynamodb:DescribeContinuousBackups",
         "dynamodb:DescribeGlobalTable"
+        "dynamodb:DescribeLimits"
+        "dynamodb:DescribeStream"
+        "dynamodb:GetRecords"
+        "dynamodb:GetShardIterator"
+        "dynamodb:ListStreams"
```
[the remark this answers is not in the document]
Comment (ericsorenson) [reply policy.json.tpl]: Updated.

File: shared/src/business/useCases/getUploadPolicyUrlInteractor.test.js [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+        isFileExists: () => false,
+      }),
+    };
+    const url = await getUploadPolicyInteractor({
+      applicationContext,
+    });
+    expect(url).toEqual('policy');
+  });
+
+  it('returns the expected policy', async () => {
```
[the remark this answers is not in the document]
Comment (ericsorenson) [reply shared/src/business/useCases/getUploadPolicyUrlInteractor.test.js]: Updated.

File: web-client/src/presenter/actions/navigateToOrdersNeededAction.test.js [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+
+  beforeEach(() => {
+    routeStub = sinon.stub();
+
+    presenter.providers.router = {
+      route: routeStub,
+    };
+  });
+
+  it('navigates to orders needed summary url when given docketNumber', async () => {
```
[the remark this answers is not in the document]
Comment (ericsorenson) [reply web-client/src/presenter/actions/navigateToOrdersNeededAction.test.js]: Updated.

File: web-client/src/presenter/actions/setPDFPageForSigningAction.test.js [hunk 6]
```diff
@@ -17,4 +17,16 @@ describe('setPDFPageForSigningAction', () => {
 
     expect(result.state.pdfForSigning.pageNumber).toEqual(newState);
   });
+
+  it('uses the default value if no pageNumber is provided', async () => {
```
[the remark this answers is not in the document]
Comment (ericsorenson) [reply web-client/src/presenter/actions/setPDFPageForSigningAction.test.js]: Updated.

File: web-client/src/presenter/computeds/trialSessionWorkingCopyHelper.test.js [hunk 7]
```diff
@@ -379,12 +391,58 @@ describe('trial session working copy computed', () => {
       ]);
       expect(result.casesShownCount).toEqual(5);
     });
+
+    it('should return all sessions if none of the case statuses are closed', () => {
```
[the remark this answers is not in the document]
Comment (ericsorenson) [reply web-client/src/presenter/computeds/trialSessionWorkingCopyHelper.test.js]: The description was inaccurate, it's been updated.

File: .dockerignore (hunk 2 above)
Comment (waldoj) [reply .dockerignore]: Works great — thank you!

**Comment by waldoj** [comment]
@JessicaMarine, I have reviewed this pull request, and recommend that the Court accept it into `staging`.
